### PR TITLE
Add setup_python_configuration_options

### PR DESCRIPTION
**Pull request by 2maz**
and allow to activate python

### CODE REVIEW

File: rock/python.rb [hunk 1]
```diff
@@ -200,6 +206,12 @@ def self.activate_python(ws: Autoproj.workspace,
         [bin, version]
     end
 
+    def self.deactivate_python(ws: Autoproj.workspace)
+        remove_python_shims(ws.root_dir)
+        ws.config.reset('python_executable')
+        ws.config.reset('python_version')
```
Comment (doudou) [inline rock/python.rb]: Should maybe save after removing the options.

File: rock/python.rb [hunk 2]
```diff
@@ -216,4 +228,26 @@ def self.activate_python_path(pkg,
                              "python#{version}","site-packages")
         [bin, version, path]
     end
+
+    def self.setup_python_configuration_options(ws: Autoproj.workspace)
+        ws.config.declare 'USE_PYTHON', 'boolean',
+            :default => 'no',
```
Comment (doudou) [inline rock/python.rb]: Use "new" (2.0+) keyword arguments (default: 'no')

File: rock/python.rb [hunk 3]
```diff
@@ -216,4 +228,26 @@ def self.activate_python_path(pkg,
...
+            :doc => [ "Do you want to activate python?" ]
+
+        value = ws.config.get('USE_PYTHON')
+        if value && !Autoproj::BuildOption::FALSE_STRINGS.include?(value)
```
Comment (doudou) [inline rock/python.rb]: 'boolean' should take care of the conversion. Doesn't it ?

Comment (2maz) [reply rock/python.rb]: It does not for autoproj <= v.2.11.0 when using noninteractive mode. Fortunately the bug has already been fixed for upcoming releases https://github.com/rock-core/autoproj/commit/efcd09dcd918281a3041b39a8f6defa3e2d7f2ea

File: rock/python.rb (hunk 1 above)
Comment (2maz) [reply rock/python.rb]: Sorry, I do not understand what you are trying to say here,

Comment (doudou) [reply rock/python.rb]: Sorry, I thought I replied.

I am wondering if it wouldn't make sense to save the configuration file right away (`ws.config.save`). Just a wonder, really.

File: rock/python.rb (hunk 3 above)
Comment (doudou) [reply rock/python.rb]: K. I need to release that stuff.

File: rock/python.rb (hunk 1 above)
Comment (2maz) [reply rock/python.rb]: That would make an exemption from the rule - in all other cases config.save is triggered after all packages sets have been loaded. So I would not do it.

**Pull request merged**